dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -250,11 +262,11 @@ def register_new_user(user)
     move_oauth_params_to_cache(user)
     session["devise.user_attributes"] = user.attributes
 
-    # For some providers, signups can happen without ever having hit the sign_up page, where
-    # our tracking data is usually populated, so do it here
-    SignUpTracking.begin_sign_up_tracking(session)
-
-    redirect_to new_user_registration_url
+    if SignUpTracking.new_sign_up_experience?(session)
+      redirect_to users_finish_sign_up_url
+    else
+      redirect_to new_user_registration_url
+    end
```

Comment:
The actual behavior split is here, although the decision about which study group the user is in happens in `begin_sign_up_tracking`.